torch/distributed/fsdp/symbolic_trace.py

```diff
@@ -0,0 +1,198 @@
+import functools
+import torch
+from dataclasses import dataclass
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    List,
+    Optional,
+    Set,
+    Tuple,
+)
+
+
+__all__ = ["TracingConfig"]
+
+
+@dataclass
+class TracingConfig:
+    """
+    Configurations used in ParamExecOrderWrapPolicy for symbolic tracing of a model.
+    tracer: An instance of torch.fx.Tracer that will be used to perform symbolic
+        tracing. tracer is default to be torch.fx.Tracer(), but can also be instance
+        of some child class of torch.fx.Tracer. For example, for hugginface transformer
+        based models, one may want to use HFTracer:
+        https://github.com/huggingface/transformers/blob/main/src/transformers/utils/fx.py#L636
```

Comment:
We should change this to a permalink independent of the `main` branch or else the link will become deprecated over time.